src/main/frontend/pipeline-console-view/pipeline-console/main/components/checkboxes.tsx

```diff
@@ -0,0 +1,227 @@
+import React from "react";
+import {
+  MainViewVisibility,
+  StageViewPosition,
+  useLayoutPreferences,
+} from "../providers/user-preference-provider.tsx";
+import "./checkboxes.scss";
+
+export default function VisibilitySelect() {
+  const {
+    mainViewVisibility,
+    setMainViewVisibility,
+    stageViewPosition,
+    setStageViewPosition,
+  } = useLayoutPreferences();
+
+  const handleViewChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setMainViewVisibility(e.target.value as MainViewVisibility);
+  };
+
+  const handlePositionChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setStageViewPosition(e.target.value as StageViewPosition);
+  };
+
+  return (
+    <>
+      {/* View Selector */}
+      <label
+        className="jenkins-dropdown__item idk"
+        htmlFor="main-view-visibility"
+      >
+        <div className="jenkins-dropdown__item__icon">
+          <ViewIcon
+            mainViewVisibility={mainViewVisibility}
+            stageViewPosition={stageViewPosition}
+          />
+        </div>
+        Views
+        <span>
+          {mainViewVisibility === MainViewVisibility.BOTH && "Stage and tree"}
+          {mainViewVisibility === MainViewVisibility.STAGE_ONLY && "Stage"}
+          {mainViewVisibility === MainViewVisibility.TREE_ONLY && "Tree"}
```

Comment:
I'm not sold on `Tree` being user facing terminology, `Console` is what its been called before what do you think?

Reply:
Wasn't a fan of _Console_ as it's too similar to _Logs_. Lewis mentioned calling them _Graph_ and _Stages_ and I quite like the sound of that. Alternative to _Graph_ would be _Pipeline_.